torch/csrc/autograd/variable.cpp

```diff
@@ -87,6 +89,37 @@ std::shared_ptr<Function> VariableImpl::get_grad_accumulator() {
   return result;
 }
 
+VariableViewImpl::VariableViewImpl(Variable base_, at::Tensor data_)
+  : VariableImpl(std::move(data_))
+  , base(std::move(base_))
+  , expected_version(0) {
+  if (!base.defined()) {
+    throw std::runtime_error("base is undefined");
+  }
+  if (base.is_view()) {
+    base = base.base();
+  }
+  is_view = true;
+  version_counter = base.version_counter();
+  expected_version = version_counter.current_version();
+}
+
+std::shared_ptr<Function>& VariableViewImpl::get_grad_fn() {
+  std::lock_guard<std::mutex> lock(grad_accumulator_lock);
+  if (expected_version != version_counter.current_version()) {
+    auto fn = std::make_shared<generated::AsStridedBackward>();
+    fn->self_geometry = TensorGeometry(base);
+    fn->size = sizes();
+    fn->stride = strides();
+    fn->storage_offset = Variable(this, true).storage_offset();
+    fn->set_flags(Function::flags({ base }));
+    fn->num_inputs = 1;
+    _grad_fn = std::move(fn);
+    expected_version = version_counter.current_version();
```

Comment:
It seems that there might be a race in here (you read some version, but it changes while you're updating the attrs), although I think it would only happen if someone modifies the base Variable, while you're reading it, which is an error itself. Is that right?

Also, it might be better to get `current_version` once, since it's an atomic load